src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,113 @@
+package parser;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.internal.bind.util.ISO8601Utils;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+import com.github.javafaker.Faker;
+import java.io.*;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JsonParserTest {
+
+    private final JsonParser jsonParser = new JsonParser();
+    private Faker faker = new Faker();
+    private final String fileName = String.valueOf(faker.name());
+    private final String filePath = "./src/main/resources/%s.json";
+
+
+
+
+    @Disabled("Disabled test")
+    @Tag("JSTest")
+    @Test
+    public void writeEmptyCartTest() { //  В этом тесте идут значения в разнобой, как я писал ранее
+        Cart testCart = new Cart("test-cart");
+        jsonParser.writeToFile(testCart);
+        try {
+            JSONObject jsonObject = new JSONObject();
+            JSONArray realItems = new JSONArray();
+            JSONArray virtualItems = new JSONArray();
+            jsonObject.put("cartName","test-cart");
+            jsonObject.put("realItems",realItems);
+            jsonObject.put("virtualItems",virtualItems);
+            jsonObject.put("total",0.0);
```

Comment:
I really can't answer on your question, what algorithm is used by json-simple library to put elements. I can recommend to use Gson library. Also, pay your attention, that you must form expected result using testCasrt object.